frontend-react/src/__mocks__/DBMock.test.ts

```diff
@@ -0,0 +1,239 @@
+import { SetupServer, setupServer } from "msw/node";
+
+import { RSOrganizationSettings } from "../config/endpoints/settings";
+import { createDbMock } from "../utils/MSWData/CreateDbMock";
+import {
+    RelativePathMap,
+    ParentResolverMap,
+    factoryRestHandlerTupleToMap,
+    RestHandlerMethods,
+} from "../utils/MSWData/EnhancedFactory";
+
+import { createModels } from "./DBMock";
+
+let { db, reset } = createDbMock(createModels);
+
+describe("DBMock", () => {
+    beforeEach(() => {
+        reset();
+    });
+
+    describe("createMany", () => {
+        const createManyScenarios: {
+            label: string;
+            args: [number, Partial<RSOrganizationSettings>[] | undefined];
+        }[] = [
+            {
+                label: "predictably random",
```

Comment:
This is subjective, but I think these sorts of tests would benefit from being broken out into their own blocks rather than using `test.each`.  I realize that this is just checking to make sure the records are created according to the specified numbers and initial data, but it's not entirely clear at first glance in my opinion -- especially with the `initial.length === 0` check (which really should be contained in its own test case).

Reply:
Overall, my opinion is that a small amount of redundancy in tests for the sake of clarity is a good thing.  DRY can be used if there's a significant benefit to it, but it feels like we're adding extra logic (nested loops + `initial.length` check) in the test cases to get around DRY rather than benefitting entirely from it.